src/cljs/athens/page.cljs

```diff
@@ -70,14 +70,40 @@
        [:div {:style {:margin-left 20}}
         [render-blocks (:block/uid @node)]]])))
 
+(def enter-keycode 13)
+(def esc-keycode 27)
+
+(defn title-comp [title]
+  (let [s (reagent/atom {:editing false
+                         :current-title title})
+        save! (fn [new-title]
+                (dispatch [:node/rename (:current-title @s) new-title])
+                (swap! s assoc :editing false))
```

Comment:
I'm wondering if we should reorder the dispatch and swap. The dispatch slows down the swap (re-rendering of the element) and might give a laggy user experience? If you try putting something too short in like "a" the whole interface stalls because of the unlinked reference feature. This is mentioned here #62 . Reversing this order here might give a more direct experience and move some of the computation in the background (of the user)